exp/ipam/api/v1beta1/ipaddressclaim_types.go

```diff
@@ -25,6 +25,10 @@ import (
 
 // IPAddressClaimSpec is the desired state of an IPAddressClaim.
 type IPAddressClaimSpec struct {
+	// ClusterName is the name of the Cluster this object belongs to.
+	// +optional
+	ClusterName string `json:"clusterName"`
```

Comment:
Should we add validation in the webhook to verify the name is a DNS domain like, or fail to create

Reply:
Maybe worth a follow-up to do this for all ClusterName fields in all our CRDs

Reply:
Not sure whether it's worth the effort tbh. It either needs to be a valid cluster name, or ti is useless. We can't validate that a cluster with that name exists, as that's very annoying with eventual consistency and so on. As long as it doesn't lead to fatal errors when looking for a cluster with that name it doesn't really matter whether the provided name _could_ be used as a cluster name or not.